packages/trip-details/src/trip-detail.tsx

```diff
@@ -11,61 +12,70 @@ type Props = {
   summary: ReactElement | string;
 };
 
-type State = {
-  expanded: boolean;
-};
+/**
+ * Copied from https://stackoverflow.com/questions/50940640/how-to-determine-if-jest-is-running-the-code-or-not
+ */
+function isRunningJest() {
+  return process.env.JEST_WORKER_ID !== undefined;
+}
 
-export default class TripDetail extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      expanded: false
-    };
-  }
+function uuidv4(): string {
+  if (isRunningJest()) return "mocked-random-id";
 
-  toggle = (): void => {
-    const { expanded } = this.state;
-    if (expanded) this.onHideClick();
-    else this.onExpandClick();
-  };
+  return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function(c) {
+    // eslint-disable-next-line no-bitwise
+    const r = (Math.random() * 16) | 0;
+    // eslint-disable-next-line no-bitwise
+    const v = c === "x" ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+  });
+}
 
-  onExpandClick = (): void => {
-    this.setState({ expanded: true });
-  };
+const TripDetail = ({ icon, summary, description }: Props): JSX.Element => {
+  const intl = useIntl();
+  const [expanded, setExpanded] = useState(false);
 
-  onHideClick = (): void => {
-    this.setState({ expanded: false });
+  const id = uuidv4();
+
+  const toggle = (): void => {
+    setExpanded(!expanded);
   };
 
-  render(): ReactElement {
-    const { icon, summary, description } = this.props;
-    const { expanded } = this.state;
-    return (
-      <S.TripDetail role="group">
-        <S.TripDetailIcon role="presentation">{icon}</S.TripDetailIcon>
-        <S.TripDetailSummary>
-          {summary}
-          {description && (
-            <S.ExpandButton id="expand-button" onClick={this.toggle}>
-              <QuestionCircle size="0.92em" />
-            </S.ExpandButton>
-          )}
-        </S.TripDetailSummary>
-        <AnimateHeight duration={300} height={expanded ? "auto" : 0}>
-          <S.TripDetailDescription
+  return (
+    <S.TripDetail role="group">
+      <S.TripDetailIcon role="presentation">{icon}</S.TripDetailIcon>
+      <S.TripDetailSummary>
+        {summary}
+        {description && (
+          <S.ExpandButton
+            aria-label={intl.formatMessage({
+              id: expanded
+                ? "otpUi.TripDetails.hideDetail"
+                : "otpUi.TripDetails.showDetail"
+            })}
```

Comment:
Separate the two strings (i.e. move the `expanded` condition outside of `intl.formatMessage` (so that CLI tools can pick up the string).